docs/design/auto-storage-scale.md

```diff
@@ -0,0 +1,49 @@
+# Automatic Storage Scaling
+
+# Summary
+
+This feature would help users to get rid of maintenance task of keeping the system with enough available free space.
+In virtualized environments, we can scale the storage devices OSDs so this feature helps in automatic scaling. 
+
+# Proposal
+
+Implement auto storage scale CR per device class of a specific Storagecluster.
+
+# High-Level Design
+
+- Create a AutoStorageScale CR per Storagecluster device class
+    ```
+    apiVersion: ocs.openshift.io/v1
+    Kind: AutoStorageScaling
+    Metadata:
+	    Name: <storageclusterName>-<deviceClassName>
+	    namespace: <namespace>
+    Spec: 
+	    storageClusterName: <storageclusterName>
+        deviceClassName: <deviceClassName>
+	    capacityLimit: <capacity>   
+    Status:
+	    phase: (waiting/inProgress/completed/failure)
+        LastRunTimeStamp: (only on completed/failure)
+   	    knownOSDSize: 2TiB
+        expectedOSDSize: 4TiB
+        knownOSDCount: 3​ 
+        expectedOSDCount: 3
+    ```
+
+-  Watch prometheus periodically after 10 mins and depending on the OSD fill percentage, recocnile the CR. 
+
+# Low-Level Design 
+
+1) Add a go-routine to watch for Prometheus metrics
+2) Go routine updates the channel and the sync-map with osd percentage
```

Comment:
What channel and sync-map is this referring to? I thought it would just update the status on the AutoStorageScaling CR?

Reply:
We need a map that will act as a source of truth instead of status